src/System.Text.Json/src/System/Text/Json/Node/JsonNodeOptions.cs

```diff
@@ -0,0 +1,109 @@
+        /// </exception>
+        /// <remarks>
+        ///   By default <exception cref="JsonException"/> is thrown if a comment is encountered.
+        /// </remarks>
+        public JsonCommentHandling CommentHandling
+        {
+            readonly get => _commentHandling;
+            set
+            {
+                Debug.Assert(value >= 0);
+                if (value > JsonCommentHandling.Skip)
+                    throw new ArgumentOutOfRangeException(nameof(value), SR.JsonDocumentDoesNotSupportComments);
+
+                _commentHandling = value;
+            }
+        }
+
+        /// <summary>
+        ///   Gets or sets the maximum depth allowed when parsing JSON data,
+        ///   with the default (that is, 0) indicating a maximum depth of 64.
+        /// </summary>
+        /// <exception cref="ArgumentOutOfRangeException">
+        ///   The max depth is set to a negative value.
+        /// </exception>
+        /// <value>
+        ///   The maximum depth allowed when parsing JSON data.
+        /// </value>
+        /// <remarks>
+        ///   Parsing past this depth will throw a <exception cref="JsonException"/>.
+        /// </remarks>
+        public int MaxDepth
+        {
+            readonly get => _maxDepth;
+            set
+            {
+                if (value < 0)
+                    throw ThrowHelper.GetArgumentOutOfRangeException_MaxDepthMustBePositive(nameof(value));
+
+                _maxDepth = value;
+            }
+        }
+
+        /// <summary>
+        ///   Gets or sets a value that indicates whether an extra comma at the end of a list of JSON values
+        ///   in an object or array is allowed (and ignored) within the JSON payload being read.
+        /// </summary>
+        /// <returns>
+        ///   <see langword="true"/> if an extra comma at the end of a list of JSON values in an object or array is allowed;
+        ///   otherwise, <see langword="false"/>. Default is <see langword="false"/>.
+        /// </returns>
+        /// <remarks>
+        ///   By default, it's set to false, and <exception cref="JsonException"/> is thrown if a trailing comma is encountered.
+        /// </remarks>
+        public bool AllowTrailingCommas { get; set; }
+
+        private DuplicatePropertyNameHandlingStrategy _duplicatePropertyNameHandlingStrategy;
+
+        /// <summary>
+        ///   Gets or sets the duplicate property name handling strategy.
+        /// </summary>
+        /// <remarks>
+        ///   By default, it's set to <exception cref="DuplicatePropertyNameHandlingStrategy.Replace"/>.
+        /// </remarks>
+        public DuplicatePropertyNameHandlingStrategy DuplicatePropertyNameHandling
+        {
+            readonly get => _duplicatePropertyNameHandlingStrategy;
+            set
+            {
+                if ((uint)value > (uint)DuplicatePropertyNameHandlingStrategy.Error)
```

Comment:
We should add tests for invalid `DuplicatePropertyNameHandling` enum values but this can be addressed separately as its test enhancement to improve coverage. 

See https://github.com/dotnet/corefx/pull/41030#discussion_r323528688
> Someone could pass in -1 for this OR an integer value larger than `DuplicatePropertyNameHandlingStrategy.Error`.

Reply:
Filed https://github.com/dotnet/corefx/issues/42585 to track this.